rllib/models/experimental/tf/primitives.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.models.specs.checker import (
+    input_is_decorated,
+    is_output_decorated,
+)
+from ray.rllib.models.temp_spec_classes import TensorDict
+from ray.rllib.models.experimental.base import Model
+from ray.rllib.utils.typing import TensorType
+from typing import Tuple
+
+_, tf, _ = try_import_tf()
+
+
+def _call_not_decorated(input_or_output):
+    return (
+        f"forward not decorated with {input_or_output} specification. Decorate "
+        f"with @check_{input_or_output}_specs() to define a specification. See "
+        f"BaseModel for examples."
+    )
+
+
+class TFModel(Model):
+    """Base class for RLlib models.
+
+    This class is used to define the general interface for RLlib models and checks
+    whether inputs and outputs are checked with `check_input_specs()` and
+    `check_output_specs()` respectively.
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+        assert input_is_decorated(self.__call__), _call_not_decorated("input")
+        assert is_output_decorated(self.__call__), _call_not_decorated("output")
+
+    def __call__(self, input_dict: TensorDict) -> Tuple[TensorDict, List[TensorType]]:
+        """Returns the output of this model for the given input.
+
+        Args:
+            input_dict: The input tensors.
+
+        Returns:
+            Tuple[TensorDict, List[TensorType]]: The output tensors.
+        """
+        raise NotImplementedError
+
+
+class FCNet(tf.Module):
```

Comment:
Can we rename this TfFCNet for clarity? And also the counterpart? `TorchFCNet`.

Reply:
Done! (TfMLP and TfMLPModel)